sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -249,11 +241,12 @@ case class Divide(left: Expression, right: Expression)
       s"${eval2.value} == 0"
     }
     val javaType = ctx.javaType(dataType)
-    val divide = if (dataType.isInstanceOf[DecimalType]) {
+    val divide = if (dataType.isInstanceOf[DecimalType] || dataType.isInstanceOf[DoubleType]) {
       s"${eval1.value}.$decimalMethod(${eval2.value})"
     } else {
-      s"($javaType)(${eval1.value} $symbol ${eval2.value})"
+      s"($javaType)(${eval1.value} $decimalMethod ${eval2.value})"
```

Comment:
but what about the decimalMethod used in line 245 ? if we inline the operator `/` there too it gives 
```Binary numeric promotion not possible on types "org.apache.spark.sql.types.Decimal" and "org.apache.spark.sql.types.Decimal"```